src/routeGeneration/templates/express.hbs

```diff
@@ -161,7 +161,9 @@ export function RegisterRoutes(app: express.Express) {
 
                 // WARNING: This file was auto-generated with tsoa. Please do not modify it. Re-run tsoa to re-generate this file: https://github.com/lukeautry/tsoa
 
-                if (data || data === false) { // === false allows boolean result
+                if (data && data.readable) {
```

Comment:
This is a dangerous check, please make sure this only runs when data is an instance of `Readable`.

Also, please copy this to the other templates.

Reply:
Are you suggesting a guard like: `data && data instanceof Readable`? This is not the recommended way according to this SO (from 2014) https://stackoverflow.com/questions/23885095/nodejs-check-if-variable-is-readable-stream. They suggest using duck-typing so something more like:

`data && data.pipe && data._read`

Alternatively [this package](https://github.com/sindresorhus/is-stream) is an option.

I'm open to whichever, what do you prefer? @WoH

Reply:
Good points. I didn't consider handling anything except the Node ReadableStream. I'm not sure we want to handle legacy streams or arbitrarily call pipe on anything that responds to that method, but there's no perfect solution here that I'm aware of.
I'd accept a check that checks for a function called pipe and makes sure the stream is readable.

Reply:
I've gone with this check: `data && typeof data.pipe === 'function' && data.readable && typeof data._read === 'function'`.

It looks like both Hapi and Koa will automatically handle streams so no further changes should be needed for the other templates.